packages/core/ui-settings/core-ui-settings-server-internal/src/ui_settings_service.ts

```diff
@@ -94,10 +99,19 @@ export class UiSettingsService
     return {
       register: this.register,
       registerGlobal: this.registerGlobal,
+      setAllowlist: this.setAllowlist,
     };
   }
 
   public async start(): Promise<InternalUiSettingsServiceStart> {
+    if (this.allowlist) {
+      // If we are in development mode, check if all settings in the allowlist are registered
+      if (process.env.NODE_ENV !== 'production') {
+        this.validateAllowlist();
+      }
```

Comment:
You shouldn't need to read from `process.env` directly here -- we expose a proper mechanism for determining dev mode from the `Env` service in core. In this service you should be able to access it from `coreContext`:

```js
class UiSettingsService {
  constructor(coreContext) {
    this.isDev = coreContext.env.mode.dev;
  }

  async start() {
    if (this.allowlist) {
      if (this.isDev) {
        this.validateAllowlist();
      }
      // ...etc
    }
  }
}
```

Reply:
Got it, thanks for letting me know!